libraries/app/database_api.cpp

```diff
@@ -198,6 +198,24 @@ class database_api_impl : public std::enable_shared_from_this<database_api_impl>
          return tmp;
       }
 
+      const account_object* get_account_from_string( const std::string& name_or_id  ) const
+      {
+         // TODO cache the result to avoid repeatly fetching from db
```

Comment:
Possible solution: Put the result in a static, always check to see if the names/ids match. 
Caution: It is my understanding that references to db objects can turn stale. That could mean that you'll be pointing to an address that could now be non-null and not an account_object.
Question: Would keeping a copy be a better choice? While the data within it could grow stale, at least you would control its lifetime.